src/transformers/models/detr/modeling_detr.py

```diff
@@ -0,0 +1,2239 @@
+    logits: torch.FloatTensor = None
+    pred_boxes: torch.FloatTensor = None
+    auxiliary_outputs: Optional[List[Dict]] = None
+    decoder_hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+    decoder_attentions: Optional[Tuple[torch.FloatTensor]] = None
+    cross_attentions: Optional[Tuple[torch.FloatTensor]] = None
+    encoder_last_hidden_state: Optional[torch.FloatTensor] = None
+    encoder_hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+    encoder_attentions: Optional[Tuple[torch.FloatTensor]] = None
+
+
+@dataclass
+class DetrForPanopticSegmentationOutput(DetrObjectDetectionOutput):
+    """
+    This class adds one attribute to DetrObjectDetectionOutput, namely predicted masks.
+
+    Args:
+        pred_masks (:obj:`torch.FloatTensor` of shape :obj:`(batch_size, ...)`):
+            ...
+    """
+
+    pred_masks: torch.FloatTensor = None
+
+
+## BELOW: utilities copied from
+# https://github.com/facebookresearch/detr/blob/master/backbone.py
+
+
+class FrozenBatchNorm2d(nn.Module):
+    """
+    BatchNorm2d where the batch statistics and the affine parameters are fixed.
+
+    Copy-paste from torchvision.misc.ops with added eps before rqsrt, without which any other models than
+    torchvision.models.resnet[18,34,50,101] produce nans.
+    """
+
+    def __init__(self, n):
+        super(FrozenBatchNorm2d, self).__init__()
+        self.register_buffer("weight", torch.ones(n))
+        self.register_buffer("bias", torch.zeros(n))
+        self.register_buffer("running_mean", torch.zeros(n))
+        self.register_buffer("running_var", torch.ones(n))
+
+    def _load_from_state_dict(
+        self, state_dict, prefix, local_metadata, strict, missing_keys, unexpected_keys, error_msgs
+    ):
+        num_batches_tracked_key = prefix + "num_batches_tracked"
+        if num_batches_tracked_key in state_dict:
+            del state_dict[num_batches_tracked_key]
+
+        super(FrozenBatchNorm2d, self)._load_from_state_dict(
+            state_dict, prefix, local_metadata, strict, missing_keys, unexpected_keys, error_msgs
+        )
+
+    def forward(self, x):
+        # move reshapes to the beginning
+        # to make it user-friendly
+        w = self.weight.reshape(1, -1, 1, 1)
+        b = self.bias.reshape(1, -1, 1, 1)
+        rv = self.running_var.reshape(1, -1, 1, 1)
+        rm = self.running_mean.reshape(1, -1, 1, 1)
+        eps = 1e-5
+        scale = w * (rv + eps).rsqrt()
+        bias = b - rm * scale
+        return x * scale + bias
+
+
+class Backbone(nn.Module):
+    """
+    Timm convolutional backbone.
+
+    If resnet, replace nn.BatchNorm2d layers by FrozenBatchNorm2d defined above.
+
+    """
+
+    def __init__(self, name: str, train_backbone: bool, dilation: bool, return_intermediate_layers: bool):
+        super().__init__()
+
+        kwargs = {}
+        if name in ["resnet18", "resnet34", "resnet50", "resnet101"]:
+            kwargs["norm_layer"] = FrozenBatchNorm2d
```

Comment:
FYI. There are quite a few other models besides those where `norm_layer` can be used to replace BatchNorm w/ FrozenBN, but unfortunately not all of them.

Reply:
Yes, I wonder how we can make this list longer in order to support more models.